generated/nidaqmx/task.py

```diff
@@ -760,6 +760,46 @@ def read(self, number_of_samples_per_channel=NUM_SAMPLES_UNSET,
 
         return data.tolist()
 
+    def add_arguments_if_necessary(self, callback_function, expected_number_of_arguments):
+        if (callback_function.__code__.co_argcount < expected_number_of_arguments):
+            print("Not enough arguments! Adding arguments")
+            if ("every_n_samples_event_type" in parameters):
+                result = self.n_samples_event_wrapper(parameters, callback_function)
+            elif ("signal_type" in parameters):
```

Comment:
Every N and signal events pass the callback an enum, but they use a bare ctypes or int type rather than the enum classes from nidaqmx.constants. See my prototype: https://github.com/ni/nidaqmx-python/pull/474/files

> callback_method(self, EveryNSamplesEventType(every_n_samples_event_type), number_of_samples)

> callback_method(self, Signal(signal_type))